Assets/Scripts/OverworldManager.cs

```diff
@@ -1,40 +1,73 @@
-            character.transform.parent = LBG.transform;
-        }
-        else
-        {
-            GameObject character = Instantiate(character2, new Vector3(0, 0, 0), Quaternion.identity);
-            character.AddComponent(typeof(ImmediatePositionWithLocationProvider));
-            character.AddComponent(typeof(RotateWithLocationProvider));
-            character.transform.parent = LBG.transform;
-
-
-        }
-
-    }
-
-    // Update is called once per frame
-    void Update()
-    {
-        
-    }
-}
+﻿using Mapbox.Examples;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class OverworldManager : MonoBehaviour
+{
+    [SerializeField] private GameObject character1;
+    [SerializeField] private GameObject character2;
+    private GameManager gameManager;
+    private GameObject player;
+    private float MIN_RANGE = -5.0f;
+    private float MAX_RANGE = 50.0f;
+    // Start is called before the first frame update
+    void Start()
+    {
+        GameObject loader = GameObject.Find("Loader");
+        gameManager = loader.GetComponent<GameManager>();
+
+        LoadCharacter();
+        PopulateMapWithBeasties();
+    }
+
+
+    private void LoadCharacter()
+    {
+        GameObject LBG = GameObject.Find("LocationBasedGame");
+        int charType = gameManager.charType;
+        if (charType == 0)
+        {
+            player = Instantiate(character1, new Vector3(0, 0, 0), Quaternion.identity);
+        }
+        else
+        {
+            player = Instantiate(character2, new Vector3(0, 0, 0), Quaternion.identity);
+        }
+
+        player.AddComponent(typeof(ImmediatePositionWithLocationProvider));
+        player.AddComponent(typeof(RotateWithLocationProvider));
+        player.transform.parent = LBG.transform;
+        // Slime model tiny - using this to increase size
+        player.transform.localScale += new Vector3(3.0f, 3.0f, 3.0f);
+    }
+
+    private void PopulateMapWithBeasties()
+    {
+        int beastiesToSpawn = gameManager.beastiesSpawned;
+        for (int i = 0; i < beastiesToSpawn; i++)
+        {
+            InstantiateBeastie();
+        }
+
+        gameManager.beastiesSpawned -= beastiesToSpawn;
+    }
+
+    private void InstantiateBeastie()
+    {
+        Beastie beastie = BeastieFactory.Instance.GenerateBeastie();
+        float x = player.transform.position.x + GenerateRange();
+        float y = player.transform.position.y;
+        float z = player.transform.position.z + GenerateRange();
+        Instantiate(beastie, new Vector3(x, y, z), Quaternion.Euler(0.0f, UnityEngine.Random.Range(0.0f, 360.0f), 0.0f));
+    }
+
+    private float GenerateRange()
+    {
+        // Reference: https://www.youtube.com/watch?v=S827tTi8OCo&list=PL86WBCjNmqh4bDQycScKKlP0ETkjo39gG
+        float randomNum = UnityEngine.Random.Range(MIN_RANGE, MAX_RANGE);
+        int direction = UnityEngine.Random.Range(0, 1) == 0 ? 1 : -1;

```

Comment:
Looks like the direction will always be 1 here, since Random.Range for ints have exclusive max values unlike the version with floats. https://docs.unity3d.com/ScriptReference/Random.Range.html

We can update this to:
`int direction = UnityEngine.Random.Range(0, 2) == 0 ? 1 : -1;`

Reply:
Good catch! Missed that when I switched the Range to int heh.